src/backend/tests/it/utils/pocketic.rs

```diff
@@ -309,3 +313,94 @@ where
         WasmResult::Reject(error) => Err(error),
     })
 }
+
+/// A test Oisy backend canister with a shared reference to the `PocketIc` instance it is installed on.
+pub struct PicBackend {
+    pub pic: Arc<PocketIc>,
+    pub canister_id: Principal,
+}
+impl PicCanisterTrait for PicBackend {
+    fn pic(&self) -> Arc<PocketIc> {
+        self.pic.clone()
+    }
+    fn canister_id(&self) -> Principal {
+        self.canister_id.clone()
+    }
+}
+impl PicBackend {
+    /// Creates toy users with the given range of principals.
```

Comment:
Note: This is copied (and adjusted) from https://github.com/dfinity/oisy-wallet/blob/cdf4246f4943215deb6eb40c16bf8cabdf5ee8e4/src/backend/tests/it/list_users.rs#L16